modules/n1ql/pages/n1ql-language-reference/searchfun.adoc

```diff
@@ -136,6 +136,21 @@ It overrides the default mapping and is used during index selection and filterin
 The object must either have a default mapping with no type mapping, or a single type mapping with the default mapping disabled.
 For more information, refer to xref:fts:fts-creating-indexes.adoc[Creating Indexes].
 
+| `indexUUID`
+[Optional]
+| string
+| A string, containing the UUID of a full text index in the keyspace.
+This provides an index hint to the Query engine.
+If the full text index cannot be identified, an error occurs.
+
+You can use the `indexUUID` field as an alternative to the `index` field.
```

Comment:
This isn't accurate. `indexUUID` isn't an alternative to the `index` field, but can be passed in alongside the `index` field to further assert that you're executing the query on the very index. It is however optional.

Reply:
Sorry @abhinavdangeti I don't quite understand this comment, given that you can use the `indexUUID` field without the `index` field. For example:

```
SELECT META(t1).id
FROM `travel-sample` AS t1
WHERE SEARCH(t1.country, "+United +States", {"indexUUID": "472b0cad1f126fe9"});
```

Reply:
This is a covering index situation. The verification phase doesn't even run in this case.

In case of a non-covering index situation (for example query for something from the object body), the verification phase will pick up the index mapping of an FTS index - if the index name is provided. To make this further accurate, if an indexUUID is also provided, the UUID of the fetched index mapping is also verified.

Reply:
Ah, understood. Thanks.